pkg/controllers/managementuser/rbac/cluster_handler.go

```diff
@@ -67,47 +69,51 @@ func (h *clusterHandler) sync(key string, obj *v3.Cluster) (runtime.Object, erro
 
 func (h *clusterHandler) doSync(cluster *v3.Cluster) error {
 	_, err := v32.ClusterConditionGlobalAdminsSynced.DoUntilTrue(cluster, func() (runtime.Object, error) {
-		// Sync both admin types
-		for _, roleName := range []string{rbac.GlobalAdmin, rbac.GlobalRestrictedAdmin} {
-			// Do not sync restricted-admin to the local cluster as 'cluster-admin'
-			if cluster.Name == "local" && roleName == rbac.GlobalRestrictedAdmin {
-				continue
+		// For restricted admins, re-enqueue the GRBs to trigger the creation of any RBAC resources on new cluster creation.
+		grbs, err := h.grbIndexer.ByIndex(grbByRoleIndex, rbac.GlobalRestrictedAdmin)
+		if err != nil {
+			return nil, fmt.Errorf("failed to list GlobalRoleBindings for restricted-admin: %w", err)
+		}
+		for _, x := range grbs {
+			grb, _ := x.(*v3.GlobalRoleBinding)
+			h.grbController.Enqueue("", grb.Name)
+		}
+
+		grbs, err = h.grbIndexer.ByIndex(grbByRoleIndex, rbac.GlobalAdmin)
+		if err != nil {
+			return nil, fmt.Errorf("failed to list GlobalRoleBindings for global-admin: %w", err)
+		}
+
+		for _, x := range grbs {
+			grb, _ := x.(*v3.GlobalRoleBinding)
```

Comment:
Nit: I missed this earlier, but this type assertion doesn't check if the assertion failed or not. It also doesn't check if the pointer is nil or not.

This could lead to a panic if this doesn't work for some reason. I'm marking this as a nit since the indexer shouldn't have that problem, but I would prefer if we could do a:
```go
if !ok || grb == nil {
  continue
}
```